hibernate-reactive-core/src/test/java/org/hibernate/reactive/containers/H2Database.java

```diff
@@ -0,0 +1,122 @@
+		expectedDBTypeForClass.put( Boolean.class, "BOOLEAN" );
+		expectedDBTypeForClass.put( NumericBooleanType.class, "INTEGER" );
+		expectedDBTypeForClass.put( TrueFalseType.class, "CHARACTER" );
+		expectedDBTypeForClass.put( YesNoType.class, "CHARACTER" );
+		expectedDBTypeForClass.put( int.class, "INTEGER" );
+		expectedDBTypeForClass.put( Integer.class, "INTEGER" );
+		expectedDBTypeForClass.put( long.class, "BIGINT" );
+		expectedDBTypeForClass.put( Long.class, "BIGINT" );
+		expectedDBTypeForClass.put( float.class, "DOUBLE PRECISION" );
+		expectedDBTypeForClass.put( Float.class, "DOUBLE PRECISION" );
+		expectedDBTypeForClass.put( double.class, "DOUBLE PRECISION" );
+		expectedDBTypeForClass.put( Double.class, "DOUBLE PRECISION" );
+		expectedDBTypeForClass.put( byte.class, "TINYINT" );
+		expectedDBTypeForClass.put( Byte.class, "TINYINT" );
+		expectedDBTypeForClass.put( PrimitiveByteArrayTypeDescriptor.class, "BINARY VARYING" );
+		expectedDBTypeForClass.put( URL.class, "VARCHAR_IGNORECASE" );
+		expectedDBTypeForClass.put( TimeZone.class, "VARCHAR_IGNORECASE" );
+		expectedDBTypeForClass.put( Date.class, "DATE" );
+		expectedDBTypeForClass.put( Timestamp.class, "TIMESTAMP" );
+		expectedDBTypeForClass.put( Time.class, "TIME" );
+		expectedDBTypeForClass.put( LocalDate.class, "DATE" );
+		expectedDBTypeForClass.put( LocalTime.class, "time" );
+		expectedDBTypeForClass.put( LocalDateTime.class, "TIMESTAMP" );
+		expectedDBTypeForClass.put( BigInteger.class, "NUMERIC" );
+		expectedDBTypeForClass.put( BigDecimal.class, "NUMERIC" );
+		expectedDBTypeForClass.put( Serializable.class, "BINARY VARYING" );
+		expectedDBTypeForClass.put( UUID.class, "binary" );
+		expectedDBTypeForClass.put( Instant.class, "datetime" );
+		expectedDBTypeForClass.put( Duration.class, "bigint" );
+		expectedDBTypeForClass.put( Character.class, "VARCHAR_IGNORECASE" );
+		expectedDBTypeForClass.put( char.class, "VARCHAR_IGNORECASE" );
+		expectedDBTypeForClass.put( TextType.class, "text" );
+		expectedDBTypeForClass.put( String.class, "VARCHAR_IGNORECASE" );
+	}}
+
+	private String getRegularJdbcUrl() {
+		return "jdbc:h2:~/test;DATABASE_TO_UPPER=FALSE";
+	}
+
+	@Override
+	public String getJdbcUrl() {
+		return "jdbc:h2:~/test;DATABASE_TO_UPPER=FALSE";
+	}
+
+	@Override
+	public String getUri() {
+		return "h2:~/test;DATABASE_TO_UPPER=FALSE";
+	}
+
+
+	@Override
+	public String getScheme() {
+		return "h2";
+	}
+
+	@Override
+	public String getNativeDatatypeQuery(String tableName, String columnName) {
+		return "SELECT DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS COLS " +
+				"WHERE COLS.TABLE_NAME = '" + tableName + "'" +
+				"AND COLS.COLUMN_NAME= '" + columnName + "'";
+	}
+
+	@Override
+	public String getExpectedNativeDatatype(Class<?> dataType) {
+		return expectedDBTypeForClass.get( dataType );
+	}
+
+	@Override
+	public String createJdbcUrl(String host, int port, String database, Map<String, String> params) {
+		return getRegularJdbcUrl();
```

Comment:
this method is called `createJdbcUrl` because is supposed to create a valid jdbc url for the database using the method parameters. It's used by a test to check that we create the right options given the parameters.

Now, I suppose most of the time people won't care for H2 when used in embedded mode. But H2 can also be configured as a remote server. This would be a valid url: `jdbc:h2:tcp://localhost/~/test`

Reply:
So building a URL would require a little work to support the different options:
* Primary mode for H2 is embedded which uses the URL format:   "jdbc:h2:~/test"
* H2 can also be configured as a remote server.
  * EXAMPLE 1:  jdbc:h2:tcp://localhost/D:/myproject/data/project-name
  * EXAMPLE 2:  jdbc:h2:tcp://localhost/~/test
  * EXAMpLE 3:  jdbc:h2:tcp://localhost:9081/~/test
  
It would make more sense to separate into it's own H2 module and write tests in there rather than complicating it in HR test?

.. and that's without knowing the structure of what that would look like and how the tests would get integrated and run.

Reply:
For the purpose of the test I think `createJdbcUrl`  could return the format as for the EXAMPLE 3